### DISCUSSION TITLE
Request support of L2TP/PPTP in Xray-core beside the Wireguard(Features request)

### DISCUSSION THREAD

**Discussion by Saeid008** [XTLS/Xray-core#2966] [Ideas]
Dear Xray-core Developer.

As we all know, recently the Wireguard has been added and is supported by Xray-core, which is a nice feature.
I requested from xray-core developer to support l2tp/pptp as well if possible.
Because there are many router out there (in market) that only support l2tp/pptp and it will be great if we can connect to an internal server(VPS) via l2tp/pptp. on internal server inside a country there is no limitations by GFW to avoid connection via l2tp/pptp or Wireguard.
So it can be very helpful if l2tp/pptp will added to Xray-core too.

Thank you.

**Comment by mrafieefard** [comment]
+

**Comment by mistek131995** [comment]
It would be nice if you implemented this

**Comment by Saeid008** [comment]
> It would be nice if you implemented this
@mistek131995 
This is something that @RPRX or others Xray-core developer needed to decide since wireguard support added.

**Comment by Sir-MmD** [comment]
> Dear Xray-core Developer.
> 
> As we all know, recently the Wireguard has been added and is supported by Xray-core, which is a nice feature. I requested from xray-core developer to support l2tp/pptp as well if possible. Because there are many router out there (in market) that only support l2tp/pptp and it will be great if we can connect to an internal server(VPS) via l2tp/pptp. on internal server inside a country there is no limitations by GFW to avoid connection via l2tp/pptp or Wireguard. So it can be very helpful if l2tp/pptp will added to Xray-core too.
> 
> Thank you.

Hi
You can use this project:
https://github.com/Sir-MmD/RTX-VPN
It's a workaround to use L2TP/PPTP/OpenVPN/SSTP with xray core as outbound

**Comment by AseWhy** [comment]
Yes, it would be great if such a feature were available out of the box.
